### DISCUSSION TITLE
Customize react antd theme?

### DISCUSSION THREAD

**Discussion by kkhanhluu** [nrwl/nx#8049] [General]
Does someone have experience customizing react antd theme? I tried to do the following steps but it doesn't succeed: 
1. Create a new `webpack.config.js` as this [comment](https://github.com/nrwl/nx/issues/3175#issuecomment-644296842)
2. point `webpackConfig` options to newly created webpack config file
3. Tried to customize less loader in webpack config file [https://ant.design/docs/react/customize-theme#Customize-in-webpack](https://ant.design/docs/react/customize-theme#Customize-in-webpack)

**Comment by szbartnik** [comment]
I've discovered that as NX's WebPack configuration already supports LESS, it has to be enhanced whereas most of resources I found were telling me to add brand new rules to WebPack.

Maybe my ugly `webpack.config.ts` will help someone:

```typescript
import nrwlConfig from '@nrwl/react/plugins/webpack.js';
import type { Configuration } from 'webpack';

function overrideLessVariables(
  config: Configuration,
  variables: Record<string, any>
) {
  config.module.rules
    .find((r) => r.test.toString().includes('.less$'))
    .oneOf.find((x) => x.test.toString() === '/\\.less$/')
    .use.find((l) =>
      l.loader.includes('less-loader')
    ).options.lessOptions.modifyVars = variables;
}

module.exports = (config: Configuration) => {
  nrwlConfig(config);
  overrideLessVariables(config, {
    'primary-color': '#eb5b1d',
    'table-expanded-row-bg': 'white',
  });
};
```
